pkg/callout/callout.go

```diff
@@ -21,6 +20,27 @@ import (
 
 var errFailedToRead = errors.New("failed to read callout response body")
 
+// Can be used in the implementation of
+// callout service with structured payload.
+type BaseRequest struct {
+	Operation  string `json:"operation,omitempty"`
+	EntityType string `json:"entity_type,omitempty"`
+	EntityID   string `json:"entity_id,omitempty"`
+	CallerID   string `json:"caller_id,omitempty"`
+	CallerType string `json:"caller_type,omitempty"`
```

Comment:
Caller can interpreted as the service calling the callout (e.g., domain service calling callout , so here domain service is caller). 

What do you think about rename CallerID and CallerType to something more descriptive like RequesterID / RequesterType ?